packages/SwingSet/src/vats/comms/clist-outbound.js

```diff
@@ -77,14 +71,8 @@ export function makeOutbound(state, stateKit) {
     remote.toRemote.set(vpid, rpid);
     remote.fromRemote.set(flipRemoteSlot(rpid), vpid);
 
-    if (p.resolved) {
-      // we must send the resolution *after* the message which introduces it
-      const { remoteID } = remote;
-      Promise.resolve().then(() =>
-        resolveToRemote(remoteID, [[vpid, p.rejected, p.data]]),
-      );
-    } else {
-      // or arrange to send it later, once it resolves
+    if (!p.resolved) {
```

Comment:
Will this new version correctly handle the case where a previously-resolved Promise is included in the arguments of a `send`? I can see how the `resolve` case will work better without this extra `resolveToRemote()`, but I suspect the `send` case really needs it.

We talked a while ago about moving this sort of "send extra messages" code out of the translation function and into a post-translation clause that looks at the `slots`, figures out which ones are Promises, figures out which of *those* are resolved and new, and then sends the necessary `resolve` messages in the same turn as the `send`. I think I'm liking that option more and more.

Reply:
This makes sense.  The kernel does that, and we're finding a lot of the comms vat logic mirrors the kernel logic.